tests/backup/backup_restore_basic_test.go

```diff
@@ -3712,3 +3713,327 @@ var _ = Describe("{DeleteS3ScheduleAndCreateNfsSchedule}", func() {
 		CleanupCloudSettingsAndClusters(backupLocationMap, s3CloudCredName, s3CloudCredUID, ctx)
 	})
 })
+
+// checkBackupObjectForNonExpectedNS checks if namespaces like kube-system and px namespace
+// is backed up or not
+func checkBackupObjectForNonExpectedNS(ctx context.Context, backupName string) {
+
+	var namespacesToSkip = []string{"kube-system", "kube-node-lease", "kube-public"}
+
+	// Fetch a list of backups
+	backupUID, err := Inst().Backup.GetBackupUID(ctx, backupName, orgID)
+	dash.VerifyFatal(err, nil, "Fetch backup UID")
+	backupInspectRequest := &api.BackupInspectRequest{
+		Name:  backupName,
+		Uid:   backupUID,
+		OrgId: orgID,
+	}
+	backupResponse, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+	dash.VerifyFatal(err, nil, "Fetch backup inspect object")
+	backupNamespaces := backupResponse.GetBackup().Namespaces
+
+	err = SetDestinationKubeConfig()
+	dash.VerifyFatal(err, nil, "Switch cluster context")
+	k8sCore := core.Instance()
+	allServices, err := k8sCore.ListServices("", metav1.ListOptions{})
+	dash.VerifyFatal(err, nil, "Get list of services")
+	for _, svc := range allServices.Items {
+		if svc.Name == "portworx-service" {
+			namespacesToSkip = append(namespacesToSkip, svc.Namespace)
+		}
+	}
+	for _, namespace := range backupNamespaces {
+		for _, namespacetoskip := range namespacesToSkip {
+			dash.VerifyFatal(namespacetoskip == namespace, false, fmt.Sprintf("Expected namespace %s shouldn't be present in backup", namespace))
+		}
+	}
+	err = SetSourceKubeConfig()
+	log.FailOnError(err, "Switching context to source cluster failed")
+}
+
+// getNamespaceAge gets the namespace age of all the namespaces on the cluster
+func getNamespaceAge() map[string]time.Time {
```

Comment:
I feel we should take namespaces as input rather than switching clusters internally.

Reply:
How about we return error?

Reply:
The intend of these functions is to reduce code duplication in case of the specific case, returning error and again adding dash.verifyfatel will come back to the same problem of redundant code. Yes it would be 3-4 redundant lines then some 10-20 lines but I feel this way it's much more cleaner.

Reply:
if we add it to the ValidateBackup func the thing this logic is only needed in case of all namespace backup and an in-place restore of all the namespaces with replace set to true. It will just make it slow for us to use the same.

Reply:
Better not because if we switch in ValidateBackup many tests will fail. Thanks.